careerfairsystems/arkad-app-2018
# Implement api and redux

## Change description

- Implementing the API for fetching companies
- Adding redux
  - Creating a filter reducer that is persisted in redux (used for favorites tracking)
- Creating a row for companies
  - Adding a temporary favorites button to it and connecting it to redux
- Creating a show favorites button
  - Connecting it to redux

## How to verify

Run the project and
- Check that pressing Not fav on a company row turns it to Fav
  - Verify that reloading the app does not reset the state
- Check that the Show favorites button filters the list to show only favorites and changes its text to Hide favorites
  - Verify that reloading the app does not reset the state
- Try changing the URL of the API so that an error is thrown, confirming that the error view is working
  - Test the reload button

## Issues fixed

This fixes #10, fixes #16, fixes #15, fixes #22.
